lua/nui/text/init.lua

```diff
@@ -35,6 +35,19 @@ function Text:set(content, extmark)
   return self
 end
 
+---@param extmark? string|table highlight group name or extmark options
+---@return NuiText
+function Text:set_highlight(extmark)
+  if extmark then
+    self.extmark = is_type("string", extmark) and { hl_group = extmark } or vim.deepcopy(extmark)
+    self.extmark.id = nil
+  else
+    self.extmark = { id = nil, hl_group = nil }
+  end
+
+  return self
+end
+
```

Comment:
I don't remember it being here last time I reviewed. 🤔

Is this a related change? Why is this needed?

Reply:
It was there previously. For complex borders, the `chars` is an array of `Text` that we want to change the highlight. I exposed `Text:set_highlight` rather than accessing internal state.

Reply:
Isn't it the same as `text:set(text:content(), highlight)`?

Reply:
If we change the `Text:init` a little bit, we can even do `text = Text(text, new_hl)`

Reply:
I feel like adding a `set_highlight` is preferrable due to its increased readbility.

Regarding `text:set(text:content(), new_hl)`: it has the disadvantage that half the characters aren't related to the action in question in this use-case, in addition to the unneeded argument passing.
Regarding `text = Text(text:content(), new_hl)`: I don't like the unneeded memory allocation; the fact that luajit is fast and has a proper GC doesn't mean we shouldn't help it, in particular in what is meant to be a core library.

Reply:
Well, this particular use-case is an internal one. And doing `text:set(text:content(), new_hl)` in this PR is not really doing any harm.

In future if we see that changing only the highlight for an existing text is a very common use-case, we can add that method later.

We don't have to do it right now, since it's not really necessary for this PR.